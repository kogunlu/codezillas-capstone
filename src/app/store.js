import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import bookingReducer from '../features/booking/bookingSlice';
import userReducer from '../features/user/userSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    booking: bookingReducer,
    user: userReducer,
  },
});
