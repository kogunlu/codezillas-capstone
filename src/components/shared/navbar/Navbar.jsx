import {React,useState} from 'react';
import {Link } from 'react-router-dom';
import Button from '../button/Button';



function Navbar() {
  const [SideMenuClose, setSideMenuClose] = useState("hidden")
  const [SideMenu, setSideMenu] = useState()
  const [ShowHide, setShowHide] = useState()


  function handleSideMenu()
  {
    if(SideMenuClose === 'hidden')
    {
      setSideMenu('hidden')
      setSideMenuClose()
      setShowHide('z-[-1] left-[0px] opacity-100 ')
    }else{
      setSideMenu()
      setSideMenuClose('hidden')
      setShowHide()
    }
  }
  return (
    <div className='nav-bg'>
      <div className='p-3'>
        <div className='md:grid md:grid-cols-4 justify-center'>
          <div className='grid grid-cols-3 md:grid-col-2 col-span-2'>
            <Link to="/">
              <span className='grid grid-cols-2 ml-3 sm:ml-3 md:ml-8  text-2xl sm:text-3xl md:text-5xl gap-1 sm:gap-4 lg:gap-0 cursor-pointer'>
                <img className='w-min inline' src="/logo.png" alt='Healing logo' />
                <span>Healing</span>
              </span>
            </Link>
            <button type='button' className='cursor-pointer md:hidden block col-start-3 grid grid-flow-col grid-cols-5 grid-rows-5' onClick={()=> handleSideMenu() } >
                <svg className={`col-start-5 row-start-2 row-span-3 ${SideMenuClose}`} xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24">
                    <g id="close">
                        <path id="x" d="M18.717 6.697l-1.414-1.414-5.303 5.303-5.303-5.303-1.414 1.414 5.303 5.303-5.303 5.303 1.414 1.414 5.303-5.303 5.303 5.303 1.414-1.414-5.303-5.303z"/>
                    </g>
                </svg>
                <svg className={`col-start-5 row-start-2 row-span-3 ${SideMenu}`} height="32px"><path d="M4,10h24c1.104,0,2-0.896,2-2s-0.896-2-2-2H4C2.896,6,2,6.896,2,8S2.896,10,4,10z M28,14H4c-1.104,0-2,0.896-2,2  s0.896,2,2,2h24c1.104,0,2-0.896,2-2S29.104,14,28,14z M28,22H4c-1.104,0-2,0.896-2,2s0.896,2,2,2h24c1.104,0,2-0.896,2-2  S29.104,22,28,22z"/></svg>
            </button>
          </div>
          <ul className={`grid place-content-center md:grid-cols-6 my-0 col-span-2 content-center z-[1] md:z-auto absolute md:static nav-bg left-0 w-full md:w-auto py-4 ma:py-0 pl-7 md:pl-0 opacity-0 md:opacity-100 left-[-400px] transition-all ease-in duration-500 gap-4 ${ShowHide}` }>
            <li className='py-2 text-center'><Link className='hover:text-yellow-white duration-300' to="/">Home</Link></li>
            <li className='py-2 text-center'><Link className='hover:text-yellow-white duration-300' to="/blogs">Blogs</Link></li>
            <li className='py-2 text-center'><Link className='hover:text-yellow-white duration-300' to="/about">About Us</Link></li>
            <li className='py-2 text-center'><Link className='hover:text-yellow-white duration-300' to="/contact">Contact Us</Link></li>
            <li className="text-center"><Button className="col-span-2 duration-500" type='button' name="Login" classes="py-2 px-5  hover:bg-cyan-200 bg-cyan-400 font-medium rounded text-sm md:text-lg"/></li>
          </ul>
        </div>
      </div>
    </div>
  )
}

export default Navbar