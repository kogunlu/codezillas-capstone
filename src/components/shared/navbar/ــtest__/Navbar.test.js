import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Navbar from '../Navbar';
import { store } from '../../../../app/store';

test('Navbar component renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
