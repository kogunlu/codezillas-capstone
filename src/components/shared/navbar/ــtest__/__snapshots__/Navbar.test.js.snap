// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly 1`] = `
<div
  className="bg-main-background"
>
  <div
    className="p-3"
  >
    <div
      className="grid grid-cols-4"
    >
      <div
        className="grid grid-cols-2 md:grid-col-2 col-span-2"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <span
            className="grid grid-cols-2 ml-3 sm:ml-3 md:ml-8  text-3xl md:text-5xl gap-1 sm:gap-4 lg:gap-0 cursor-pointer"
          >
            <img
              alt="Healing logo"
              className="w-min inline"
              src="/logo.png"
            />
            <span>
              Healing
            </span>
          </span>
        </a>
      </div>
      <button
        className="cursor-pointer md:hidden block col-start-4 grid grid-flow-col grid-cols-4 grid-rows-5"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="col-start-5 row-start-2 row-span-3 text-3xl hidden"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3,3 L21,21 M3,21 L21,3"
            fill="none"
            stroke="#000"
            strokeWidth="2"
          />
        </svg>
        <svg
          className="col-start-5 row-start-2 row-span-3 w-8 text-3xl "
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2,19 L22,19 M2,5 L22,5 M2,12 L22,12"
            fill="none"
            stroke="#000"
            strokeWidth="2"
          />
        </svg>
      </button>
      <ul
        className="grid place-content-center md:grid-cols-12 mt-16 md:my-0 col-span-2 content-center  md:z-auto absolute md:static bg-main-background left-0 w-full md:w-auto py-4 ma:py-0 pl-7 md:pl-0  md:opacity-100  transition-all ease-in duration-500 gap-4 left-[-400px] opacity-0 z-[-1]"
      >
        <li
          className="py-2 text-center md:col-span-2"
        >
          <a
            aria-current="page"
            className="hover:text-yellow-white duration-300 active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="py-2 text-center md:col-span-2"
        >
          <a
            className="hover:text-yellow-white duration-300"
            href="/blogs"
            onClick={[Function]}
          >
            Blogs
          </a>
        </li>
        <li
          className="py-2 flex justify-center grid md:col-span-3"
        >
          <button
            className="hover:text-yellow-white duration-300 grid grid-cols-3 w-30 "
            onClick={[Function]}
            type="button"
          >
            <span
              className="col-span-2"
            >
              About
            </span>
            <span>
              <svg
                className="text-xl"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  fill="none"
                  points="18 9 12 15 6 9"
                  stroke="#000"
                  strokeWidth="2"
                />
              </svg>
            </span>
          </button>
          <ul
            className="row-span-2 bg-white grid gap-3 px-8 divide-y divide-slate-300 border-2 border-slate-400 mt-10 static md:block md:absolute transition-all ease-in duration-500 z-[1] opacity-0 hidden"
          >
            <li
              className="text-center py-2 w-full"
            >
              <a
                className="hover:text-yellow-white duration-300"
                href="/about"
                onClick={[Function]}
              >
                About Us
              </a>
            </li>
            <li
              className="text-center py-2 w-full"
            >
              <a
                className="hover:text-yellow-white duration-300"
                href="/team"
                onClick={[Function]}
              >
                Team
              </a>
            </li>
            <li
              className="text-center py-2  w-full"
            >
              <a
                className="hover:text-yellow-white duration-300"
                href="/career"
                onClick={[Function]}
              >
                Career
              </a>
            </li>
          </ul>
        </li>
        <li
          className="py-2 text-center md:col-span-2"
        >
          <a
            className="hover:text-yellow-white duration-300"
            href="/contact"
            onClick={[Function]}
          >
            Contact Us
          </a>
        </li>
        <li
          className=" md:col-span-3 grid place-content-center w-32"
        >
          <input
            className="py-2 px-5  hover:bg-cyan-200 bg-cyan-400 font-medium rounded text-sm md:text-lg"
            onClick={[Function]}
            type="button"
            value="Login"
          />
          <ul
            className="row-span-2 bg-white grid gap-3 px-8 divide-y divide-slate-300 border-2 border-slate-400 mt-12 static md:block md:absolute transition-all ease-in duration-500 z-[1] opacity-0 hidden"
          >
            <li
              className="text-center py-2 w-full hover:text-yellow-white duration-300 "
            >
              <button
                className="place-self-center"
                onClick={[Function]}
                type="button"
              >
                Sign Out
              </button>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
